src/common/type_defs.h

```diff
@@ -0,0 +1,34 @@
+/*
+
+   Copyright 2023 Leil Storage OÜ
+
+   This file is part of SaunaFS.
+
+   SaunaFS is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, version 3.
+
+   SaunaFS is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with SaunaFS. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "common/platform.h"
+
+/// Inode is meant to be used as type for all inode number references.
+/// Needs to be compatible with C++ and C code.
+#ifdef __cplusplus
+#include <cstdint>
+
+using Inode = uint32_t;
```

Comment:
Consider naming the type as `Inode_t`
```suggestion
using Inode_t = uint32_t;
```

Reply:
Thanks, yes, I will do it in later commits.